src/pydap/handlers/dap.py

```diff
@@ -143,18 +147,29 @@ def __getitem__(self, index):
             int(np.ceil((s.stop-s.start)/float(s.step))) for s in index)
         size = int(np.prod(shape))
 
-        if self.dtype == np.byte:
-            return np.fromstring(data[:size], 'B')
-        elif self.dtype.char in 'SU':
-            out = []
-            for word in range(size):
-                n = np.fromstring(data[:4], '>I')  # read length
-                data = data[4:]
-                out.append(data[:n])
-                data = data[n + (-n % 4):]
-            return np.array([ text_type(x.decode('ascii')) for x in out ], 'S')
-        else:
-            return np.fromstring(data, self.dtype).reshape(shape)
+        try:
+            if self.dtype == np.byte:
+                return np.fromstring(data[:size], 'B')
+            elif self.dtype.char in 'SU':
+                out = []
+                for word in range(size):
+                    n = np.fromstring(data[:4], '>I')  # read length
+                    data = data[4:]
+                    out.append(data[:n])
+                    data = data[n + (-n % 4):]
+                return np.array([ text_type(x.decode('ascii')) for x in out ], 'S')
+            else:
+                return np.fromstring(data, self.dtype).reshape(shape)
+        except ValueError as e:
```

Comment:
I've moved it closer to the reshape, which is the operation causing the error. I don't know why I've put it so far away. Thanks for pointing this out.